cpp/ps0004/problem 5.cpp

```diff
@@ -1,26 +1,34 @@
-# include <iostream>
+#include <iostream>
 using namespace std ;
 int main ()
 {
-float number; 
+    // prompting user for number 1.5/1.5
+    float number; 
     cout << "Enter a number (must be between 32 and 126): ";
     cin >> number;
-    if ( number - int (number) != 0)
-  {  // إذا كان المدخل غير صحيح(رقم عشري)
+
+    // checking invalidaty of user's input 4/4
+	// printing the number and the opposite printable character 2/2
+    if (number - int(number) != 0)
+    {  // إذا كان المدخل غير صحيح(رقم عشري)
         cout << "Invalid: a floating point number " << endl;
     }
     
-   else if (number == 0 || cin.fail() ){
+    else if (number == 0 || cin.fail() ){
         // إاذا كان المدخل غير صحيح(مثلا حرف أو صفر)
         cout << "Invalid: character or zero value" << endl;
-        }
-      else if ( number < 32 || number > 126 || number < 0)
-      {//إذا كان المدخل سالب او خارج الرينج
-      cout << "Invalid number is out of the range" << endl;   }
-       else {
+    }
+    
+    else if ( number < 32 || number > 126)
+    {//إذا كان المدخل سالب او خارج الرينج
+        cout << "Invalid number is out of the range" << endl;
+    }
+    
+    else {
         // طباعة الرمز المقابل في جدول ASCII
         cout << "The character for " << number << " is " << char(number) << endl;
     }
 
     return 0;
-}
\ No newline at end of file
+}
+// great work 😎 10.5/10 ⭐
```

Comment:
- [x] program is in the correct path
- [x] no syntax errors, bugs or warnings while compilation and execution
- [x] formatted code to be clear and readable
- [x] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
+1 bonus mark for your effort ⭐

you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more